source/grid/grid_tools.cc

```diff
@@ -491,6 +482,60 @@ namespace GridTools
         return false;
       }
     };
+
+
+    /**
+     * get_coarse_mesh_description() needs to store face data for
+     * dim>1, but we can not have this code in the function, as this
+     * requires and instantiation of CellData<dim-1> or constexpr
+     * if. Use a class instead for now.
+     */
+    template <int dim>
+    class FaceDataHelper
+    {
+    public:
+      template <class FaceIteratorType>
+      void
+      add(const FaceIteratorType &face)
```

Comment:
I'd call this function `insert_face_data()` or something similar that described in more concrete terms what exactly is being added -- because it's not the `face` itself.